### DISCUSSION TITLE
Version 0.12 is releases

### DISCUSSION THREAD

**Discussion by s-allius** [s-allius/tsun-gen3-proxy#261] [Announcements]
I am proud to present version 0.12, which was released just in time for Christmas. The main featueres are:

1. The Proxy as a Home Assistant add-on
2. A new Emulator Mode for the Solarman Protocol
3. Inverter Events 

Details of the features follow below.

I am very happy that Michael (@mime24) supported me so much with the new version. Thanks to his efforts, the proxy will now also be released as a Home Assistant add-on. This is certainly a big step to make the proxy easier to use.
However, we can still use support, for example for the creation of instructions and FAQs.

Best regards Stefan

---
## The Proxy as a Home Assistant add-on

For all those who use Home Assistant (HA), the proxy is now also available as an add-on. However, the prerequisite is that
Home Assistant is running in the Home Assistant OS or Supervised version. Container and Core installations do not support add-ons.

For an easy start there is a [guide](https://github.com/s-allius/ha-addons/blob/main/ha_addon_rel/DOCS.md) here

---
## A new Emulator Mode for the Solarman Protocol

The GEN3PLUS inverters have been using an encrypted connection to the cloud for some time now. This is good, but prevents the data from being used locally and also prevents commands from the cloud from being tracked or blocked.

Proxy users can access the inverter data using the polling mode. However, if you want to continue using the TSUN app, you must accept the encrypted connection to the TSUN Cloud and thus relinquish control over your inverters.
This is where - comes into play.

The emulator mode enables the proxy to establish a connection to the TSUN Cloud. To do this, it uses the access data of the inverter and forwards the data from polling mode to the cloud.
The advantage is that all commands now go from the TSUN Cloud to the proxy and no longer to the inverter. This gives us control over updates and configuration changes again.

---
## Inverter Events

From version 0.12, the proxy now also understands alarms and inverter notifications. These are forwarded via MQTT and can then be analysed with Home Assistant, for example.

---
## What's Changed
* Dev 0.11 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/200
* Dev 0.11 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/201
* define __slots__ for class ByteFifo by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/202
* report alarm and fault bitfield to ha by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/204
* S allius/issue205 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/207
* Bump aiohttp from 3.10.5 to 3.10.11 in /app by @dependabot in https://github.com/s-allius/tsun-gen3-proxy/pull/209
* S allius/pytest by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/211
* Add on by @mime24 in https://github.com/s-allius/tsun-gen3-proxy/pull/212
* migrate paho.mqtt CallbackAPIVersion to VERSION2 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/225
* S allius/issue222 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/223
* S allius/issue217 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/229
* S allius/issue206 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/213
* S allius/issue217 2 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/230
* Dev 0.12 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/237
* Update dependency aiohttp to v3.11.10 by @renovate in https://github.com/s-allius/tsun-gen3-proxy/pull/240
* Update python Docker tag by @renovate in https://github.com/s-allius/tsun-gen3-proxy/pull/241
* move ha repro file into root dir by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/242
* Issue220 ha addon dokumentation update by @mime24 in https://github.com/s-allius/tsun-gen3-proxy/pull/232
* S allius/issue216 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/235
* Issue220 ha addon dokumentation update by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/245
* 238 ha addon repository check by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/244
* ignore inverter replays which a older than 1 day by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/246
* S allius/issue7 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/248
* configure log path and max number of daily log files by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/243
* Update README.md by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/251
* add german language file by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/253
* avoid forwarding to a private (lokal) IP addr by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/256
* fix client_mode configuration block by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/252
* Dev 0.12 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/257
* Dev 0.12 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/258

## New Contributors
* @dependabot made their first contribution in https://github.com/s-allius/tsun-gen3-proxy/pull/209
* @mime24 made their first contribution in https://github.com/s-allius/tsun-gen3-proxy/pull/212
* @renovate made their first contribution in https://github.com/s-allius/tsun-gen3-proxy/pull/240

**Full Changelog**: https://github.com/s-allius/tsun-gen3-proxy/compare/v0.11.0...v0.12.0

<hr /><em>This discussion was created from the release <a href='https://github.com/s-allius/tsun-gen3-proxy/releases/tag/v0.12.0'>Version 0.12</a>.</em>

**Comment by mime24** [comment]
Hey Stefan, we need a quick fix to the add-on repository. the repository must include the run.sh, otherwise the add ons are not showing up in Homeassistant.

**Reply by s-allius** [reply]
ok, I will fix it in the make files

**Reply by s-allius** [reply]
Pls check again

**Reply by mime24** [reply]
addons appear now but have installation issues

`Can't install docker.io/sallius/tsun-gen3-addon:0.12.0-rel: 404 Client Error for http+docker://localhost/v1.47/images/create?tag=0.12.0-rel&fromImage=docker.io%2Fsallius%2Ftsun-gen3-addon&platform=linux%2Farm64: Not Found ("manifest for sallius/tsun-gen3-addon:0.12.0-rel not found: manifest unknown: manifest unknown")`

**Reply by mime24** [reply]
do you have an idea to cope with this message? even if we present a prebuilt container, others usually have the dockerfile also in the repo folder.

**Reply by mime24** [reply]
I found following note in the tutorial:

> If you don't use local build on the device or our build script, make sure that the Dockerfile also has a set of labels that include:

```
LABEL \
  io.hass.version="VERSION" \
  io.hass.type="addon" \
  io.hass.arch="armhf|aarch64|i386|amd64"
```


others use e.g.

```
LABEL \
    io.hass.name="${BUILD_NAME}" \
    io.hass.description="${BUILD_DESCRIPTION}" \
    io.hass.arch="${BUILD_ARCH}" \
    io.hass.type="addon" \
    io.hass.version=${BUILD_VERSION} \
```

**Reply by s-allius** [reply]
> addons appear now but have installation issues
> 
> `Can't install docker.io/sallius/tsun-gen3-addon:0.12.0-rel: 404 Client Error for http+docker://localhost/v1.47/images/create?tag=0.12.0-rel&fromImage=docker.io%2Fsallius%2Ftsun-gen3-addon&platform=linux%2Farm64: Not Found ("manifest for sallius/tsun-gen3-addon:0.12.0-rel not found: manifest unknown: manifest unknown")`

Ohh, here is the path wrong. The release container are stored on GitHub and not on docker.io.

**Reply by s-allius** [reply]
Now the image path is fixed. Pls check again

**Reply by mime24** [reply]
you made it! proxy is running
